regipy/plugins/system/usbstor.py

```diff
@@ -0,0 +1,72 @@
+
+import logging
+
+from regipy.exceptions import RegistryKeyNotFoundException
+from regipy.hive_types import SYSTEM_HIVE_TYPE
+from regipy.plugins.plugin import Plugin
+from regipy.utils import convert_wintime
+
+logger = logging.getLogger(__name__)
+
+USBSTOR_KEY_PATH = r'Enum\USBSTOR'
+PROPERTIES_NAME_GUID = r'{540b947e-8b40-45bc-a8a2-6a0b894cbda2}'
+PROPERTIES_DATES_GUID = r'{83da6326-97a6-4088-9453-a1923f573b29}'
+
+
+class USBSTORPlugin(Plugin):
+    NAME = 'usbstor_plugin'
+    DESCRIPTION = "Parse the connected USB devices history"
+    COMPATIBLE_HIVE = SYSTEM_HIVE_TYPE
+
+    def run(self):
+        try:
```

Comment:
This try block is huge, if one of the keys does not exist in a specific control set all the plugin would fail...